xcdat/spatial_avg.py

```diff
@@ -203,6 +203,27 @@ def _validate_axis(
 
         return axis
 
+    def _validate_domain_bounds(self, domain_bounds: xr.DataArray):
```

Comment:
In refactoring this code, I realize I was making the assumption that domain bounds were ordered from low to high, (e.g., `[[0, 1], [1, 2], ..., [359, 360]]`) and not the opposite or mixed (e.g., `[[1, 0], [2, 1], ..., [360, 358]]`). I *think* this assumption is also consistent with [cf-conventions](https://cfconventions.org/cf-conventions/cf-conventions.html#cell-boundaries):

> For a coordinate variable such as lat(lat) with associated boundary variable `latbnd(x,2)`, the interval endpoints must be ordered consistently with the associated coordinate, e.g., for an increasing coordinate, `lat(1) > lat(0)` implies `latbnd(i,1) >= latbnd(i,0)` for all `i` (figure 1).

Reply:
Sounds good since it follows CF conventions.

If the bounds aren't following this convention, should the user fix the bounds before spatial averaging?
We can also add a conditional to check that all min bound values are less than all max bounds values and raise ValueError if any of them aren't.

Reply:
> If the bounds aren't following this convention, should the user fix the bounds before spatial averaging?

I think they will need to, because they will get a ValueError if their bounds are not ordered correctly when calling spatial averaging. 

> We can also add a conditional to check that all min bound values are less than all max bounds values and raise ValueError if any of them aren't.

I think this is what is done [right here](https://github.com/XCDAT/xcdat/blob/c56569440c1d79c8b578008dc087f949e9d37582/xcdat/spatial_avg.py#L220).

Reply:
> I think this is what is done right here.

That check is performed on the user specified tuple representing the region bounds, and not the domain bounds found in the Dataset. 

We can add a check here when we retrieve the domain bounds. https://github.com/XCDAT/xcdat/blob/c56569440c1d79c8b578008dc087f949e9d37582/xcdat/spatial_avg.py#L329-L338

However, we can also avoid hand-holding and let the user take responsibility for making sure bounds are CF compliant.

Reply:
Ignore all comments I have just made, this makes sense now